multidim_threshold/search.py

```diff
@@ -0,0 +1,69 @@
+import numpy as np
+import multidim_threshold
+from collections import namedtuple
+import more_itertools
```

Comment:
LOL, my old boss at mozilla wrote this library!!!

I'm cool with using it, but 2 caveats:

1. Personally, I'm a bigger fan of funcy (which is already a dependency) and also implements `flatten` (which seems to be why we have the dependency).

2. If we do add it, we need to update setup.py to reflect the dependency.